### PR TITLE
feat: add minpoly.eq_of_algHom_eq

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

From #6718 (and originally https://github.com/negiizhao/transcendental/commit/c033ba3deb71278f730f41766b5a7f114c3ac991)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/Minpoly/Field.lean [hunk 1]
```diff
@@ -126,19 +126,29 @@ theorem eq_of_irreducible [Nontrivial B] {p : A[X]} (hp1 : Irreducible p)
   · rwa [Polynomial.Monic, leadingCoeff_mul, leadingCoeff_C, mul_inv_cancel]
 #align minpoly.eq_of_irreducible minpoly.eq_of_irreducible
 
+/-- If `y` is the image of `x` under an injective algebra homomorphism, their minimal polynomials
+coincide.
+
+We take `h : y = f x` as an argument because `rw [h]` typically fails
+since `IsIntegral K y` depends on y.
```
Comment (alreadydone) [inline Mathlib/FieldTheory/Minpoly/Field.lean:133]: I find this comment weird: where does `IsIntegral K y` even appear? If it does, why can't you rewrite it by `h` too?
From [search results](https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4%20eq_of_algebraMap_eq&type=code) it seems `h = rfl` most of the time.

Comment (eric-wieser) [reply Mathlib/FieldTheory/Minpoly/Field.lean:133]: I also find it weird, but it's just a generalization of a previous lemma with this comment, so I'm happy to merge as is and let someone fix it in a follow-up.

bors d+

File: Mathlib/FieldTheory/Minpoly/Field.lean [hunk 2]
```diff
@@ -126,19 +126,29 @@ theorem eq_of_irreducible [Nontrivial B] {p : A[X]} (hp1 : Irreducible p)
...
+-/
+theorem eq_of_algHom_eq {K S T : Type _} [Field K] [Ring S] [Ring T] [Algebra K S] [Algebra K T]
+    (f : S →ₐ[K] T) (hf : Function.Injective f) {x : S} {y : T} (hx : IsIntegral K x)
+    (h : y = f x) : minpoly K x = minpoly K y :=
+  minpoly.unique _ _ (minpoly.monic hx)
+    (by rw [h, aeval_algHom_apply, minpoly.aeval, AlgHom.map_zero])
+    (fun q q_monic root_q =>
+      minpoly.min _ _ q_monic (by rwa [h, aeval_algHom_apply, map_eq_zero_iff _ hf] at root_q))
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/Minpoly/Field.lean:141]: Is there some overlap here with `minpoly_algHom`?

Comment (eric-wieser) [reply Mathlib/FieldTheory/Minpoly/Field.lean:141]: Yep, `rw [h, minpoly_algHom _ hf]` works here!

bors r-

Comment (alreadydone) [reply Mathlib/FieldTheory/Minpoly/Field.lean:141]: Nice, and we only need [CommRing K].

Comment (eric-wieser) [reply Mathlib/FieldTheory/Minpoly/Field.lean:141]: It does look like we're missing `minpoly_algebraMap`, which would follow as a special case just as you do in the PR.

Comment (alreadydone) [reply Mathlib/FieldTheory/Minpoly/Field.lean:141]: #7228

**Comment by alreadydone** [comment]
I've removed its use [here](https://github.com/leanprover-community/mathlib4/pull/6718/commits/d152b6bfb616c7a16ccac8cd09cab14da9ead579#diff-fea77002a05d31f376258763c9c8a1a8301b1a0173cfd9f84300dac2e12b0c76R242-R243)

**Pull request closed without merging**